lib/client/mcp/uri.go

```diff
@@ -130,6 +145,22 @@ func (u ResourceURI) String() string {
 	return c.String()
 }
 
+// StringWithParams returns the string representation of the resource URI
+// including the query params.
+func (u ResourceURI) StringWithParams() string {
+	return u.url.String()
+}
+
+// Equal returns true if both resources represent the same Teleport resource.
+func (u ResourceURI) Equal(b ResourceURI) bool {
```

Comment:
nit: instead of implementing the function twice (like with query and without query), maybe we could implement a function that returns a new ResourceURI without the query params like `fullURI.WithoutQuery().String() == wantedURI`.  That way we can keep `String` simple, and no need to implement `Equal`.

Reply:
I've updated the `ResourceURI` following your suggestion.